app/javascript/utilities/documentation/index.js

```diff
@@ -0,0 +1,160 @@
+  const groupedRules = rules.reduce((acc, rule) => {
+    if (rule.media) {
+      return acc;
+    }
+
+    // Utility classes can modify more than one property, so we classify the CSS rule under
+    // more than one CSS property potentially.
+    // It means things will be repeated in Storybook, but it's all auto-generated, so no biggie.
+    for (let i = 0; i < rule.style.length; i++) {
+      const cssProperty = rule.style[i];
+
+      acc[cssProperty] = acc[cssProperty] || {};
+      acc[cssProperty][rule.selectorText] = rule;
+    }
+
+    return acc;
+  }, {});
+
+  return groupedRules;
+}
+
+/**
+ * Generates the content for Storybook stories for all the CSS utility
+ * classes associated to the given CSS property.
+ *
+ * @param {string} cssProperty A CSS property
+ * @param {object} cssRules A lookup whose keys are CSS utility class
+ * names and the values are CSS rules.action-space
+ *
+ * @returns {string} The content for Storybook stories for all the CSS
+ * utility classes associated to the given CSS property
+ */
+function generateUtilityClassStories(cssProperty, cssRules) {
+  const storybookStories = [
+    `  // This is an auto-generated file. DO NOT EDIT
+    import { h } from 'preact';
+    import '../../crayons/storybook-utilities/designSystem.scss';
+
+    export default {
+      title: '5_CSS Utility classes/${cssProperty}',
+    };`,
+  ];
+
+  for (const [className, cssRule] of Object.entries(cssRules)) {
+    const sanitizedCssClassName = className.replace(/[.-]/g, '_');
+    const propertiesAndValues = [];
+    let isImportant = false;
+
+    for (let i = 0; i < cssRule.style.length; i++) {
+      const styleProperty = cssRule.style[i];
+      const value = cssRule.style[styleProperty];
+
+      if (!isImportant) {
+        isImportant = cssRule.style._importants[styleProperty] === 'important';
+      }
+
+      propertiesAndValues.push(`<li>
+          <a
+            href="https://developer.mozilla.org/en-US/docs/Web/CSS/${styleProperty}"
+            target="_blank"
+            rel="noopener noreferrer"
+          >${styleProperty}</a> set to <code>${value}</code>
+        </li>`);
+    }
+
+    storybookStories.push(`
+    export const ${sanitizedCssClassName} = () => <div class="container">
+      <p><code>${className}</code> utility class for the following CSS properties:</p>
+      <ul>
+        ${propertiesAndValues.join('')}
+      </ul>
+      ${
+        isImportant
+          ? '<p>Note that <code>!important</code> is being used to override pre-design system CSS.</p>'
+          : ''
+      }
+      <pre><code>{\`${prettier.format(cssRule.cssText, {
+        parser: 'css',
+      })}\`}</code></pre>
+    </div>
+
+    ${sanitizedCssClassName}.story = { name: '${className.replace(
+      /^\./,
+      '',
+    )}' };
+    `);
+  }
+
+  return storybookStories.join('');
+}
+
+async function generateUtilityClassesDocumentation(
+  styleSheet,
+  fileWriter = file.writeFile,
```

Comment:
Just a handy way to use the file system but also to be able to test things by passing in a mock file writer.